### ISSUE TITLE
DEPRICATED Documentation notes

### ISSUE THREAD

**Issue by Maelstromeous** [ps2alerts/api#3]
# Response codes

The API has standardized response codes. The API provides 3 messages for you, the HTTP response code (i.e `200` for OK response), a API-Centric Error code (listed below) and a hopefully meaningful message.
### Example Error Response

Request: HTTP GET "/v2/alerts/1"
###### Response:

```
{
    "error": {
        "code": "API-EMPTY",
        "http_code": 203,
        "message": "No data / Empty"
    }
}
```
#### 'API-MALFORMED-REQUEST'

The data you sent couldn't be parsed correctly. This either means you sent invalid JSON or not the parameters expected by the API.
#### 'API-NOT-FOUND'

Basically a 404. The resource endpoint was not located.
#### 'API-DOH'

Server error. Hopefully a message has been passed. If you encounter this, please make an issue.
#### 'API-UNAUTHORIZED'

Unauthorized access to the endpoint requested. **\* TO IMPLEMENT ***
#### 'API-DENIED'

Your user level is unable to access this resource. Mainly used for locking down in-development endpoints. **\* Currently nothing uses this yet ***
#### 'API-EMPTY'

The particular data you were trying to find returned empty. Either that means it doesn't exist or there was simply no data to show.

**Comment by Maelstromeous** [comment]
@microwavekonijn Please let me know if the responses listed above is still relevent

**Comment by Maelstromeous** [comment]
Fyi API will have to be namedspaced as V3, as technically the current API is version 2.

**Comment by microwavekonijn** [comment]
Seems pretty standard. Keep in mind 2 things though: REST has some pretty strict rules(up to you if you keep to them), and you don't want to expose to much info.

What I would do is just use the http status codes with a message on error. Only exception is validation where you can return a validation error object with details(e.g. what fields are incorrect).

**Comment by Maelstromeous** [comment]
Agreed, if you get a 200 with no body, everything has suceeded, unless it's a `GET`. 203 will emit an empty body as the code is sufficent to tell the client "no data buddy"

**Comment by microwavekonijn** [comment]
If there is no data it should be a 404, 203 is for requests that will have no body like delete or patch/put.
